x/genutil/types/codec.go

```diff
@@ -0,0 +1,22 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// ModuleCdc defines a generic sealed codec to be used throughout this module
+var ModuleCdc *codec.Codec
+
+// TODO: abstract genesis transactions registration back to staking
+// required for genesis transactions
+func init() {
+	ModuleCdc = codec.New()
+	stakingtypes.RegisterCodec(ModuleCdc)
```

Comment:
do we need register all these on the ModuleCdc? Looks like authtypes is the only one that is used by ModuleCdc in this module. Maybe sdk too but I don't see where

Reply:
yeah we need the staking `MsgCreateValidator` and the auth `Accounts`